src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs

```diff
@@ -76,7 +76,23 @@ protected override void ExecuteCore()
                 // matchingRTReference.GetMetadata("Profile") will be "WindowsForms". 'Profile' will be an empty string if no matching RuntimeFramework is found
                 HashSet<string> profiles = matchingRuntimeFrameworks?
                     .Where(matchingRTReference => runtimePack.GetMetadata("FrameworkName").Equals(matchingRTReference.ItemSpec))
-                    .Select(matchingRTReference => matchingRTReference.GetMetadata("Profile")).ToHashSet() ?? []; 
+                    .Select(matchingRTReference => matchingRTReference.GetMetadata("Profile")).ToHashSet() ?? [];
+
+                // Special case the Windows SDK projections. These are in a runtime pack, not a runtime framework, but they also
+                // use profiles. Right now there are no other runtime packs using profiles, so we can special case these to keep
+                // the changes to a minimum and reduce risk. This logic can be revisited later to be generalized in some way.
+                if (runtimePack.GetMetadata(MetadataKeys.FrameworkName) == "Microsoft.Windows.SDK.NET.Ref")
+                {
+                    if (FrameworkReferences?.Any(fxReference => fxReference.ItemSpec == "Microsoft.Windows.SDK.NET.Ref.Windows") == true)
+                    {
+                        profiles.Add("Windows");
+                    }
+                    
+                    if (FrameworkReferences?.Any(fxReference => fxReference.ItemSpec == "Microsoft.Windows.SDK.NET.Ref.Xaml") == true)
+                    {
+                        profiles.Add("Xaml");
+                    }
+                }
```

Comment:
The reason why we added this is because the Windows SDK projections are a runtime pack, not a framework reference (even though they still use a framework reference, I have to say I don't completely understand how this works 😅), so things weren't being loaded correctly without this special casing. We also did this as it seemed less risky than making the logic work for any runtime packs. @manodasanW do you remember more details about this specifically?

Reply:
If I remember from the binlogs, this task was previously gathering the profiles from the `RuntimeFrameworks` which we don't have any.  We have `RuntimePacks`.  And `RuntimeFrameworks` look to be different from `FrameworkReferences`.  So, this special casing was to consult the `FrameworkReferences` to build the list of supported profiles rather than just looking at the `RuntimeFrameworks` which would be empty.  Realistically it does sound like having just `RuntimePacks` is a supported scenario so the logic in this task may need some updating to handle both scenarios in the long term.

Reply:
After looking at this, I think it's OK to special case.  I've suggested an update to the comment explaining what's happening.